ToDoApp/ToDoApp/ToDoListViewModel.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  ToDoListViewModel.swift
+//  ToDoApp
+//
+//  Created by Trong Hieu Vuong on 2022-12-09.
+//
+
+import Foundation
+
+final class ToDoListViewModel: ObservableObject {
+    enum State {
+        case idle
+        case loading
+        case loaded([ToDoSection])
+        case failed(Error)
+    }
+    @Published var state: State = .idle
+    private let taskService: ToDoService = FeaturesToDo()
+    func onAppear() {
+        if case .loaded = state {
+            return
+        }
+        state = .loading
+        Task {
+            let result = await taskService.getTasks()
+            switch result {
+            case .success(let todos):
+                var todoSections: [ToDoSection] = []
+                if !todos.today.isEmpty {
+                    todoSections.append(ToDoSection(title: "Today’s To-Do List", toDoItems: todos.today))
```

Comment:
another not obvious thing about strings is that in some future we may want to localize our app to be available in different languages. but using string literal outside of swiftui views will not help us to extract those strings to localization files. while localization is a big topic to discuss, we may use NSLocalizedString
```
NSLocalizedString("Today’s To-Do List", comment: "today tasks section title")
```
and if we would have localization files we would simply need to provide a translation there without changing the source code